Integrations/integration-Tufin.yml

```diff
@@ -0,0 +1,321 @@
+            {'Start': querystring['src'], 'Devices in Path' : "-->".join(['**' + d['name'] + '**' + ' ({})'.format(d['vendor']) for d in o['path_calc_results']['device_info']]), 'End': querystring['dst']}, ['Start', 'Devices in Path', 'End'])
+        except KeyError:
+            entry['HumanReadable'] = 'Unknown Output Returned'
+            entry['Contents'] = 'Unknown Output Returned'
+            entry['ContentsFormat'] = formats['text']
+            entry['Type'] = entryTypes['error']
+            return entry
+
+        return entry
+
+
+    def path_finder_image():
+        """ Sample query: querystring = {"src":"10.80.80.0","dst":"172.16.200.80","service":"tcp:80","includeIncompletePaths":"true","displayBlockedStatus":"true"} """
+        querystring = {
+            "src": demisto.args()['Source'],
+            "dst": demisto.args()['Destination'],
+            "service": demisto.args().get('Service', 'Any'),
+            "includeIncompletePaths": "true",
+            "displayBlockedStatus": "true"
+        }
+
+        entry = {
+            'Type': entryTypes['note'],
+            'Contents': '',
+            'ContentsFormat': formats['text'],
+            'ReadableContentsFormat': formats['markdown'],
+            'HumanReadable': ''
+        }
+
+        try:
+            img = tos_request("/securetrack/api/topology/path_image", querystring, {'accept': "image/png",'content-type': "application/json",'cache-control': "no-cache"})
+            if len(img) > 100:
+                return fileResult('topo.png', img, entryTypes['image'])
+            else:
+                entry['HumanReadable'] = 'No Valid Path Found'
+                entry['Contents'] = 'No Valid Path Found'
+                return entry
+        except:
+            entry['HumanReadable'] = 'Error Running Query'
+            entry['Contents'] = 'Error Running Query'
+            entry['Type'] = entryTypes['error']
+            return entry
+
+
+    def device_name(devices, device_id):
+        return [e['name'] + ' ({} {})'.format(e['vendor'], e['model']) for e in devices if int(e['id']) == int(device_id)][0]
+
+
+    def object_lookup():
+        """ Sample query: querystring = {"filter":"subnet","count":"50","exact_subnet":"1.1.1.1"} """
+        querystring = {
+            "filter":"subnet",
+            "count":"50",
+            "exact_subnet": demisto.args()['IP']
+        }
+
+        entry = {
+            'Type': entryTypes['note'],
+            'Contents': '',
+            'ContentsFormat': formats['json'],
+            'ReadableContentsFormat': formats['markdown'],
+            'HumanReadable': ''
+        }
+
+        return_json = {"objects": []}
+
+        o = tos_request("/securetrack/api/network_objects/search", querystring)
+
+        if o['network_objects']['count'] > 0:
```

Comment:
It might be a bit risky to access dictionary keys like that, as sometimes they won't exist there.
Please use `.get()`

Reply:
Agreed, moved to try in next commit